Libraries/PeriphDrivers/Source/SPIMSS/spimss_reva.c

```diff
@@ -231,6 +241,203 @@ int MXC_SPIMSS_RevA_MasterTrans(mxc_spimss_reva_regs_t *spi, spimss_reva_req_t *
+}
+
+static mxc_dma_config_t dma_config_tx;
+static mxc_dma_config_t dma_config_rx;
+static mxc_dma_adv_config_t dma_adv_config_tx;
+static mxc_dma_adv_config_t dma_adv_config_rx;
+static mxc_dma_srcdst_t srcdst_config_tx;
+static mxc_dma_srcdst_t srcdst_config_rx;
+/* ************************************************************************** */
+int MXC_SPIMSS_RevA_MasterTransDMA(mxc_spimss_reva_regs_t *spi, spimss_reva_req_t *req)
+{
+    // Calculating the transaction size in byte.
+    int sent_byte_len = ((req->len * (req->bits / 8)) - req->tx_num);
+    int tx_channel_id = -1;
+    int rx_channel_id = -1;
+    int spi_num = MXC_SPIMSS_GET_IDX((mxc_spimss_regs_t *)spi);
+    int ret_val = E_NO_ERROR;
+
+    //Reset static variables.
+    memset(&dma_config_tx, 0, sizeof(dma_config_tx));
+    memset(&dma_config_rx, 0, sizeof(dma_config_rx));
+    memset(&srcdst_config_tx, 0, sizeof(srcdst_config_tx));
+    memset(&srcdst_config_rx, 0, sizeof(srcdst_config_rx));
+    memset(&dma_adv_config_tx, 0, sizeof(dma_adv_config_tx));
+    memset(&dma_adv_config_rx, 0, sizeof(dma_adv_config_rx));
+
+    //We use SPIMSS master trans function to send data from
+    ret_val = MXC_SPIMSS_RevA_TransSetup(spi, req, 1);
+    if (E_NO_ERROR != ret_val) {
+        return ret_val;
+    }
+
+    //Setting dmas fifo levels for tx and rx in spimss.
+    spi->dma &= ~(MXC_F_SPIMSS_DMA_RX_FIFO_LVL);
+    spi->dma &= ~(MXC_F_SPIMSS_DMA_TX_FIFO_LVL);
+
+    //Initialization of DMA
+    if (true == states[spi_num].auto_dma_handlers) {
+        MXC_DMA_Init();
+    }
+
+    if ((true == states[spi_num].auto_dma_handlers) && (0 > states[spi_num].channelTx)) {
+        //Getting the available tx channel id.
+        tx_channel_id = MXC_DMA_AcquireChannel();
+        if (tx_channel_id < 0) {
+            return tx_channel_id;
+        }
+        MXC_SPIMSS_RevA_SetTXDMAChannel(spi, tx_channel_id);
+
+    } else {
+        tx_channel_id = MXC_SPIMSS_RevA_GetTXDMAChannel(spi);
+        if (tx_channel_id < 0) {
+            return E_BAD_STATE;
+        }
+    }
+    //Enable NVIC interrupt of the dma channel which will be used as TX channel.
+    MXC_NVIC_SetVector(MXC_DMA_CH_GET_IRQ(tx_channel_id), MXC_DMA_Handler);
+
+    if ((true == states[spi_num].auto_dma_handlers) && (0 > states[spi_num].channelRx)) {
+        rx_channel_id = MXC_DMA_AcquireChannel();
+        if (rx_channel_id < 0) {
+            return rx_channel_id;
+        }
+        MXC_SPIMSS_RevA_SetRXDMAChannel(spi, rx_channel_id);
+    } else {
+        rx_channel_id = MXC_SPIMSS_RevA_GetRXDMAChannel(spi);
+        if (rx_channel_id < 0) {
+            return E_BAD_STATE;
+        }
+    }
+    //Enable NVIC interrupt of the dma channel which will be used as RX channel.
+    MXC_NVIC_SetVector(MXC_DMA_CH_GET_IRQ(rx_channel_id), MXC_DMA_Handler);
+
+    states[spi_num].req = req;
+
+    // TX dma configuration settings.
+    dma_config_tx.ch = tx_channel_id;
+    dma_config_tx.reqsel = MXC_DMA_REQUEST_SPI1TX;
```

Comment:
I can forsee this being a problem for the MAX32650 because the SPIMSS DMA request select value on that part is different to the MAX32660 (`MXC_DMA_REQUEST_SPIMSSTX`, not `MXC_DMA_REQUEST_SPI1TX`). Due to this naming complication, we generally select the proper DMA request somewhere separately, typically within the part-specific library files (e.g. `spimss_me10.c` and `spimss_me11.c`).

We'll have to pull out this setting from `spimss_reva.c`, and either 1) use a new function to set the request select separately, or 2) update the input parameters of `MXC_SPIMSS_RevA_MasterTransDMA(...)` in `spimss_reva.c` to pass in the appropriate DMA request select values selected from `MXC_SPIMSS_MasterTransDMA(...)` in `spimss_me11.c`.

For example, here is how our SPI DMA drivers approach this:
https://github.com/Analog-Devices-MSDK/msdk/blob/main/Libraries/PeriphDrivers/Source/SPI/spi_me11.c#L242-L284

https://github.com/Analog-Devices-MSDK/msdk/blob/main/Libraries/PeriphDrivers/Source/SPI/spi_reva1.c#L930-L1107

Reply:
I simply added these macro controls:

For tx:
#if TARGET_NUM == 32660
    dma_config_tx.reqsel = MXC_DMA_REQUEST_SPI1TX;
#elif TARGET_NUM == 32650
    dma_config_tx.reqsel = MXC_DMA_REQUEST_SPIMSSTX;
#else
    return E_NOT_SUPPORTED;
#endif

For rx:
#if TARGET_NUM == 32660
    dma_config_rx.reqsel = MXC_DMA_REQUEST_SPI1RX;
#elif TARGET_NUM == 32650
    dma_config_tx.reqsel = MXC_DMA_REQUEST_SPIMSSRX;
#else
    return E_NOT_SUPPORTED;
#endif

Reply:
@sihyung-maxim MAX32660 enum using `SPI1TX` seems inconsistent.  Can we change it to match the UG?  Request select and all the SPIMSS registers use `SPIMSS` as well.

![image](https://github.com/Analog-Devices-MSDK/msdk/assets/38844790/3d9ac8ef-1f93-416e-b2ed-cd6103168285)

Reply:
Sure. We can't rename the `DMA_CFG.reqsel` register definitions yet in the user guide or MSDK, but for now, I added the SPIMSS (`MXC_DMA_REQUEST_SPIMSSRX/TX`) DMA request select definitions in `dma.h`.

I opened a PR in your fork @kilicomercan which should target this PR's branch. https://github.com/kilicomercan/msdk/pull/1

Reply:
I merged the PR @sihyung-maxim and removed macro(if) checks in spimss_reva.c since enumerations are defined with the same name for both microcontrollers(MAX32660 and MAX32650). 

However, we get auto build error for this PR because of some changes in main branch but not because of the changes in this PR @Jake-Carter .
![image](https://github.com/Analog-Devices-MSDK/msdk/assets/134288264/0e0337ee-c515-4dfa-8fe0-76d3f569e393)

I think that this happens because some structs defined in tft_ili9341.h are not included for CAM02_RevA.
![image](https://github.com/Analog-Devices-MSDK/msdk/assets/134288264/e76509f7-8ca3-4dee-a0a7-b137de709b09)

Reply:
Thanks @kilicomercan, fixed in https://github.com/Analog-Devices-MSDK/msdk/pull/892.  Once it's merged you can pull from main again and the tests should pass

Reply:
Thanks @Jake-Carter , I pulled from main and rebased the repository for clean commit history. No error from auto tests :)